proxy/envoy/net.go

```diff
@@ -95,17 +70,3 @@ func getPrivateIPIfAvailable() net.IP {
 	}
 	return net.IPv4zero
 }
-
-// Returns a private IPv4 address, or unspecified IP (0.0.0.0) if no IP is available
-func getPrivateIPv4IfAvailable() net.IP {
-	addrs, _ := net.InterfaceAddrs()
-	for _, address := range addrs {
-		// check the address type and if it is not a loopback the display it
-		if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
-			if ipnet.IP.To4() != nil {
-				return ipnet.IP
-			}
-		}
-	}
-	return net.IPv4zero
-}
```

Comment:
Where is the replacement? You removed these and you are still using these functions in main

Reply:
I'm not sure using local inet is a good idea. How do we pick the right interface if there are multiple available? What about ipv6?

Reply:
He is removing this function.. See this: https://github.com/istio/pilot/pull/1120/files#diff-10b44b0c268f7471a1cd69e4e0f59cc0L80 which is more generic.

Reply:
@GregHanson I feel that this file is not envoy specific. We could probably move this into proxy/